addons/binding/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/handler/AbstractOmnilinkHandler.java

```diff
@@ -1,26 +1,88 @@
 package org.openhab.binding.omnilink.handler;
 
+import java.util.Optional;
+
 import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
 import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
 import org.eclipse.smarthome.core.thing.ThingUID;
 import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
 import org.openhab.binding.omnilink.OmnilinkBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.digitaldan.jomnilinkII.MessageTypes.statuses.Status;
 
 /**
  *
  * @author Craig Hamilton
  *
  */
-public abstract class AbstractOmnilinkHandler extends BaseThingHandler {
+public abstract class AbstractOmnilinkHandler<T extends Status> extends BaseThingHandler {
+
+    private static Logger logger = LoggerFactory.getLogger(AbstractOmnilinkHandler.class);
 
     public AbstractOmnilinkHandler(Thing thing) {
         super(thing);
     }
 
+    private volatile Optional<T> status;
+
     public OmnilinkBridgeHandler getOmnilinkBridgeHander() {
         return (OmnilinkBridgeHandler) getBridge().getHandler();
     }
 
+    @Override
+    public void initialize() {
+        Optional<T> status = retrieveStatus();
+        handleStatus(status.orElse(null)); // handle status will process null.
```

Comment:
This looks like we will be retrieving the status even if no channels are linked.  Would it be better if we retrieved status if 1 or more channels are linked?  Would that be an else clause in channelLinked on the empty check?

Reply:
I think both options have merits and drawbacks.  In the current method, if we have a channel eventually linked, it will be faster because while OpenHab is doing the mechanics of linking channels we will be retrieving the status and it will be ready to go once linked - so in theory it should be faster.
The downside is, if a channel is never linked we wasted compute/network/etc getting that data.  I don't know what percent of time we have handlers created and never link channels?

I am ok, either way.

Reply:
As you suggest, I imagine most times a handler will have channels linked.  I'd say leave it as you have it.